### PR TITLE
header-only OxenMQ RPC logger

### PR DESCRIPTION
**Pull request by tewinget**

### CODE REVIEW

**Comment by tewinget** [comment]
Could maybe put an `extern std::shared_ptr<oxen::log::RingBufferSink>` in `oxen/log/omq_logger.hpp` with a prescribed name/namespace instead of passing it as a constructor argument to the `PubsubLogger` class.

File: include/oxen/log/ring_buffer_sink.hpp [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        spdlog::memory_buf_t buf;
+        formatter_->format(msg, buf);
+        std::string as_str = to_string(buf);
+        if (onLog) onLog(as_str);
+        logs.log(std::move(as_str));
+    }
+
+    void set_log_callback(cb callback = nullptr)
+    {
+        std::lock_guard<std::mutex> lock{mutex_};
```
Comment (jagerman) [inline include/oxen/log/ring_buffer_sink.hpp]:
```suggestion
        std::lock_guard lock{mutex_};
```

File: include/oxen/log/ring_buffer_sink.hpp [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        onLog = callback;
+    }
+
+    std::list<std::string> get_all() {
+        std::lock_guard<std::mutex> lock{mutex_};
```
Comment (jagerman) [inline include/oxen/log/ring_buffer_sink.hpp]:
```suggestion
        std::lock_guard lock{mutex_};
```

File: include/oxen/log/ring_buffer_sink.hpp [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+class RingBufferSink : public sink_type
+{
+    detail::LogMessageList logs;
+
+    using cb = std::function<void(const std::string&)>;
+    cb onLog = nullptr;
+
+public:
+    RingBufferSink(size_t max_size = 100, cb callback = nullptr)
+        : logs(max_size), onLog(callback) {}
```
Comment (jagerman) [inline include/oxen/log/ring_buffer_sink.hpp]:
```suggestion
        : logs{max_size}, onLog{std::move(callback)} {}
```
Curlies + move

**Review by jagerman — APPROVED** [review]

**Review by jagerman — APPROVED** [review]

**Pull request merged**